src/lib/operators/join_nested_loop.cpp

```diff
@@ -62,17 +76,14 @@ join_two_typed_segments(const BinaryFunctor& func, LeftIterator left_it, LeftIte
 
 namespace opossum {
 
-/*
- * This is a Nested Loop Join implementation completely based on iterables.
- * It supports all current join and predicate conditions, as well as NULL values.
- * Because this is a Nested Loop Join, the performance is going to be far inferior to JoinHash and JoinSortMerge,
- * so only use this for testing or benchmarking purposes.
- */
-
 JoinNestedLoop::JoinNestedLoop(const std::shared_ptr<const AbstractOperator>& left,
                                const std::shared_ptr<const AbstractOperator>& right, const JoinMode mode,
-                               const OperatorJoinPredicate& primary_predicate)
-    : AbstractJoinOperator(OperatorType::JoinNestedLoop, left, right, mode, primary_predicate, {}) {}
+                               const OperatorJoinPredicate& primary_predicate,
+                               const std::vector<OperatorJoinPredicate>& secondary_predicates)
+    : AbstractJoinOperator(OperatorType::JoinNestedLoop, left, right, mode, primary_predicate, secondary_predicates) {
+  Assert(mode != JoinMode::AntiNullAsTrue || _secondary_predicates.empty(),
+         "AntiNullAsTrue joins are not supported by JoinHash with secondary predicates.");
```

Comment:
`JoinHash` ... even though you will tell me not to take a lot at it yet.

Reply:
ah, thanks.